File: src/test/mylang/objects/MyLangInt.java
package mylang.objects;

import org.ascore.tokens.Token;

/**
 * An example of an object for the MyLang programming main.language
 */
public record MyLangInt(Integer value) implements MyLangNumber {
    public MyLangInt(Token token) {
        this(Integer.parseInt(token.getValue()));
    }

    @Override
    public Integer getValue() {
        return value;
    }
}

File: src/resources/templates/basic/kotlin/objects/MyLangInt.kt
package mylang.objects

import org.ascore.tokens.Token

/**
 * An example of an object for the MyLang programming main.language
 */
data class MyLangInt(private val value: Int) : MyLangNumber {
    constructor(token: Token) : this(token.value.toInt()) {}

    override fun getValue(): Int = value
}